src/formatDate.js

```diff
@@ -8,7 +8,48 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [, , , fromSeparator] = fromFormat;
+  const [, , , toSeparator] = toFormat;
```

Comment:
The code assumes that the separator is always the fourth element in the format arrays. This might not be the case if the format arrays are not structured as expected. Consider validating the format arrays to ensure they contain the expected elements.